Assets/Content/Items/Functional/Generic/Power Cells/PowerSupply.cs

```diff
@@ -0,0 +1,33 @@
+using System;
+using UnityEngine;
+using Mirror;
+
+namespace SS3D.Engine.Inventory
+{
+    [Serializable]
+    public struct PowerSupply
+    {
+        [SerializeField] [SyncVar] private int charge;
+        [SerializeField] private int maxCharge;
+        [SerializeField] private int chargeRate;
+        [SerializeField] private bool selfRecharge;
+
+        public int Charge => charge;
+        public int MaxCharge => maxCharge;
+        public int ChargeRate => chargeRate;
+        public bool SelfRecharge => selfRecharge;
+
+        public PowerSupply(int charge, int maxCharge, int chargeRate, bool selfRecharge)
+        {
+            this.charge = charge;
+            this.maxCharge = maxCharge;
+            this.chargeRate = chargeRate;
+            this.selfRecharge = selfRecharge;
+        }
+
+        public PowerSupply WithCharge(int ccharge)
+        {
+            return new PowerSupply(ccharge, maxCharge, chargeRate, selfRecharge);
+        }
```

Comment:
Is there a reason why this creates a new instance?

Reply:
It's a struct, you can't call it without the new keyword

Reply:
I don't mean the new keyword, why doesn't it just change the value?

Reply:
cause this is not used only for changing the charge, it's also used to create a power cell with an charge already